pkg/platform/baremetal/baremetal.go

```diff
@@ -275,6 +279,14 @@ func (c *config) checkValidConfig() hcl.Diagnostics {
 		})
 	}
 
+	if c.DownloadProtocol != "http" && c.DownloadProtocol != "https" {
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid value for `download_protocol`",
+			Detail:   fmt.Sprintf("expected 'http' or 'https', got: %q", c.DownloadProtocol),
```

Comment:
Are those the only supported protocols? No tftp for example?

Reply:
Yes, it's based on iPXE as first payload, tftp is not used at this point anymore.

Reply:
https://en.wikipedia.org/wiki/IPXE#Boot_manager says FTP and NFS can be used as well.

Reply:
Yes, but not with the Flatcar image server nor with the local matchbox asset server.